### PR TITLE
Feature/backchannel authentication support

### PR DESCRIPTION
**Pull request by travisspencer**
back-channel support

### CODE REVIEW

File: test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java [hunk 1]
```diff
@@ -60,34 +64,50 @@ public boolean startAuthentication(String authReqId,
...
+            {
+                Instant startAuthInstant = mutableRequestSubjectMap.get(authReqId).getValue();
+                Instant continueInstant = startAuthInstant.plus(_backchannelConfiguration.getDelay().get(), SECONDS);
+                Instant now = Instant.now();
+
+                if (now.isBefore(continueInstant))
+                {
+                    _logger.trace("Authentication pending till {}..", continueInstant);
+
+                    return Optional.empty();
```
Comment (renatoathaydes) [inline test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java]: I think you should return `STARTED` (i.e. the current status). Returning `empty` I think has the effect of telling the controller the status is not known.

File: test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java [hunk 2]
```diff
@@ -89,11 +84,18 @@ else if ("denying-user".equals(subject))
                 if (now.isBefore(continueInstant))
                 {
                     _logger.trace("Authentication pending till {}..", continueInstant);
-
                     return createStartedBackchannelAuthenticationResult();
                 }
             }
 
+            Optional<String> mayBeRejectingUser = _backchannelConfiguration.getRejectingUser();
```
Comment (travisspencer) [inline test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java]: this should be `maybeRejectingUser`.

File: test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
+package io.curity.identityserver.plugin.test.authentication;
```
Comment (travisspencer) [inline test-authenticator/src/main/java/io/curity/identityserver/plugin/test/authentication/TestBackchannelAuthenticatorHandler.java:17]: copyright and license header

File: test-authenticator/src/main/java/io/curity/identityserver/plugin/test/config/TestBackchannelAuthenticatorConfig.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package io.curity.identityserver.plugin.test.config;
```
Comment (travisspencer) [inline test-authenticator/src/main/java/io/curity/identityserver/plugin/test/config/TestBackchannelAuthenticatorConfig.java:17]: same here

File: test-authenticator/src/main/java/io/curity/identityserver/plugin/test/descriptor/TestBackchannelAuthenticatorPluginDescriptor.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package io.curity.identityserver.plugin.test.descriptor;
```
Comment (travisspencer) [inline test-authenticator/src/main/java/io/curity/identityserver/plugin/test/descriptor/TestBackchannelAuthenticatorPluginDescriptor.java:17]: ditto

**Pull request merged**